crates/chain/src/persist.rs

```diff
@@ -1,26 +1,33 @@
-use core::convert::Infallible;
-
 use crate::Append;
+use alloc::boxed::Box;
+use anyhow::Result;
+use core::fmt;
 
 /// `Persist` wraps a [`PersistBackend`] (`B`) to create a convenient staging area for changes (`C`)
 /// before they are persisted.
 ///
 /// Not all changes to the in-memory representation needs to be written to disk right away, so
 /// [`Persist::stage`] can be used to *stage* changes first and then [`Persist::commit`] can be used
 /// to write changes to disk.
-#[derive(Debug)]
-pub struct Persist<B, C> {
-    backend: B,
+pub struct Persist<C> {
+    backend: Box<dyn PersistBackend<C>>,
     stage: C,
 }
 
-impl<B, C> Persist<B, C>
+impl<C: fmt::Debug> fmt::Debug for Persist<C> {
```

Comment:
Alternatively, we can define the trait as

```rust
pub trait PersistBackend<C>: fmt::Debug {...}
```

and still be able to `#[derive(Debug)]` as before. I think you'd have to add a bound `C: fmt::Debug` for example in the trait impl for `Store`. Feel free to ignore if you think it's inconsequential.

Reply:
Just going to leave as is to keep my change minimal for this commit